pages/api/data/staratlas/markets/[marketId].ts

```diff
@@ -1,14 +1,68 @@
 import type { NextApiRequest, NextApiResponse } from 'next'
+import { Connection, PublicKey } from '@solana/web3.js'
+import { Market } from '@project-serum/serum'
+import { getSortedRoutes } from 'next/dist/shared/lib/router/utils'
 
-type Data = {
-    name: string
-}
 
-export default function handler(
+const CONNECTION = new Connection('https://solana-api.projectserum.com')
+// mainnet program id
+const SERUM_V3_PROGRAM_ADDRESS = new PublicKey("9xQeWvG816bUx9EPjHmaT23yvVM2ZWbrrpZb9PusVFin")
+
+function randomIntFromInterval(min : number, max : number) { 
+    return Math.floor(Math.random() * (max - min + 1) + min)
+}
+  
+  
+export default async (
     req: NextApiRequest,
-    res: NextApiResponse<Data>
-) {
+    res: NextApiResponse
+) => {
     const marketId = req.query.marketId as string
-    res.status(200).json({ name: marketId })
-    // TODO: this should return data on the specific marketId
+
+    let marketAddress = new PublicKey(marketId)
+    let market = await Market.load(CONNECTION, marketAddress, {}, SERUM_V3_PROGRAM_ADDRESS)
+    const fills = await market.loadFills(CONNECTION)
```

Comment:
I think you used to have error handling here? if this fails we need to return a 400 with an error message or we'll just explode with a 500 later down this function

Reply:
yeah, was testing and missed it. I have re-added it and testing is good!